dev/tools/gen_defaults/lib/color_scheme_template.dart

```diff
@@ -0,0 +1,86 @@
+  ColorSchemeTemplate(super.blockName, super.fileName, super.tokens);
+
+  // Map of light color scheme token data from tokens.
+  late Map<String, dynamic> colorTokensLight = tokens['colorsLight'] as Map<String, dynamic>;
+
+  // Map of dark color scheme token data from tokens.
+  late Map<String, dynamic> colorTokensDark = tokens['colorsDark'] as Map<String, dynamic>;
+
+  @override
+  String generate() => '''
+const ColorScheme _colorSchemeLightM3 = ColorScheme(
+  brightness: Brightness.light,
+  primary: Color(${tokens[colorTokensLight['md.sys.color.primary']]}),
+  onPrimary: Color(${tokens[colorTokensLight['md.sys.color.on-primary']]}),
+  primaryContainer: Color(${tokens[colorTokensLight['md.sys.color.primary-container']]}),
+  onPrimaryContainer: Color(${tokens[colorTokensLight['md.sys.color.on-primary-container']]}),
+  secondary: Color(${tokens[colorTokensLight['md.sys.color.secondary']]}),
+  onSecondary: Color(${tokens[colorTokensLight['md.sys.color.on-secondary']]}),
+  secondaryContainer: Color(${tokens[colorTokensLight['md.sys.color.secondary-container']]}),
+  onSecondaryContainer: Color(${tokens[colorTokensLight['md.sys.color.on-secondary-container']]}),
+  tertiary: Color(${tokens[colorTokensLight['md.sys.color.tertiary']]}),
+  onTertiary: Color(${tokens[colorTokensLight['md.sys.color.on-tertiary']]}),
+  tertiaryContainer: Color(${tokens[colorTokensLight['md.sys.color.tertiary-container']]}),
+  onTertiaryContainer: Color(${tokens[colorTokensLight['md.sys.color.on-tertiary-container']]}),
+  error: Color(${tokens[colorTokensLight['md.sys.color.error']]}),
+  onError: Color(${tokens[colorTokensLight['md.sys.color.on-error']]}),
+  errorContainer: Color(${tokens[colorTokensLight['md.sys.color.error-container']]}),
+  onErrorContainer: Color(${tokens[colorTokensLight['md.sys.color.on-error-container']]}),
+  background: Color(${tokens[colorTokensLight['md.sys.color.background']]}),
+  onBackground: Color(${tokens[colorTokensLight['md.sys.color.on-background']]}),
+  surface: Color(${tokens[colorTokensLight['md.sys.color.surface']]}),
+  onSurface: Color(${tokens[colorTokensLight['md.sys.color.on-surface']]}),
+  surfaceVariant: Color(${tokens[colorTokensLight['md.sys.color.surface-variant']]}),
+  onSurfaceVariant: Color(${tokens[colorTokensLight['md.sys.color.on-surface-variant']]}),
+  outline: Color(${tokens[colorTokensLight['md.sys.color.outline']]}),
+  outlineVariant: Color(${tokens[colorTokensLight['md.sys.color.outline-variant']]}),
+  shadow: Color(${tokens[colorTokensLight['md.sys.color.shadow']]}),
+  scrim: Color(${tokens[colorTokensLight['md.sys.color.scrim']]}),
+  inverseSurface: Color(${tokens[colorTokensLight['md.sys.color.inverse-surface']]}),
+  onInverseSurface: Color(${tokens[colorTokensLight['md.sys.color.inverse-on-surface']]}),
+  inversePrimary: Color(${tokens[colorTokensLight['md.sys.color.inverse-primary']]}),
+  surfaceTint: Color(${tokens[colorTokensLight['md.sys.color.primary']]}),
```

Comment:
Is this a typo? Shouldn't this be:

```suggestion
  surfaceTint: Color(${tokens[colorTokensLight['md.sys.color.surface-tint']]}),
```

Reply:
Thanks a lot for the suggestions! Fixing others, and for this one, in the token files(color_light and color_dark), the "md.sys.color.surface-tint" is mapped to "primary" and I'm not able to find a "primary" color value in palette.json. So I thought the surfaceTint color here should be the same as the primary color?

Reply:
Oh, I see. I vaguely remember something about this when they introduced the surfaceTint. Not sure why they aren't have it it be a full reference to 'md.sys.color.primary'. That said, your solution looks right. We might want to put a comment in the template as to why this is so that future us will know this was intentional and not an oversight.

Reply:
Got it! Adding a comment here:)